File: rust/src/api_internal/destination.rs
// this file is @generated
use super::DestinationAutoconfig;
use crate::Configuration;

pub struct Destination<'a> {
    cfg: &'a Configuration,
}

impl<'a> Destination<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    pub fn autoconfig(&self) -> DestinationAutoconfig<'a> {
        DestinationAutoconfig::new(self.cfg)
    }
}

File: javascript/src/api_internal/destination.ts
// this file is @generated

import { DestinationAutoconfig } from "./destinationAutoconfig";
import type { SvixRequestContext } from "../request";

export class Destination {
  public constructor(private readonly requestCtx: SvixRequestContext) {}

  public get autoconfig() {
    return new DestinationAutoconfig(this.requestCtx);
  }
}
